app/uk/gov/hmrc/servicecommissioningstatus/service/StatusCheckService.scala

```diff
@@ -0,0 +1,240 @@
+    } yield StatusCheck(status.nonEmpty, evidence)
+  }
+
+  private def checkAppConfigExistsForEnv(serviceName: String, environment: Environment)(implicit hc: HeaderCarrier): Future[StatusCheck] = {
+    for {
+      resp     <- gitHubConnector.getGithubRaw(s"/hmrc/app-config-${environment.asString}/main/$serviceName.yaml")
+      status    = resp.filter(_.status == 200)
+      evidence  = status.map(_ => s"https://github.com/hmrc/app-config-$environment/blob/main/$serviceName.yaml")
+    } yield StatusCheck(status.nonEmpty, evidence)
+  }
+
+  private def checkAlertConfigExists(serviceName: String, inputStream: InputStream): StatusCheck = {
+
+    val githubLink = "https://github.com/hmrc/alert-config/tree/main/src/main/scala/uk/gov/hmrc/alertconfig/configs"
+    val zip = new ZipInputStream(inputStream)
+
+    Iterator.continually(zip.getNextEntry)
+      .takeWhile(z => z != null)
+      .foldLeft(StatusCheck(status = false, None))((found, entry) => {
+        entry.getName match {
+          case n if n.equals(s"target/output/configs/$serviceName.json") => StatusCheck(status = true, Some(githubLink))
+          case _                                                         => found
+        }
+      })
+  }
+
+  private def checkIsDeployedForEnv(serviceName: String, releases: Seq[Release], env: Environment): StatusCheck = {
+    if (releases.map(_.environment).contains(env.asString)) {
+      StatusCheck(status = true, Some(s"https://catalogue.tax.service.gov.uk/deployment-timeline?service=$serviceName"))
+    } else {
+      StatusCheck(status = false, None)
+    }
+  }
+
+  private def checkKibanaDashboardExists(serviceName: String)(implicit hc: HeaderCarrier): Future[StatusCheck] = {
+    for {
+      inputStream <- gitHubConnector.streamGithubCodeLoad("/hmrc/kibana-dashboards/zip/refs/heads/main")
+      zip = new ZipInputStream(inputStream.get)
+      statusCheck =  Iterator.continually(zip.getNextEntry)
+        .takeWhile(z => z != null)
+        .filter(_.getName.contains("src/main/scala/uk/gov/hmrc/kibanadashboards/digitalservices"))
+        .map {_.getName -> Source.fromInputStream(zip).getLines().mkString("\n")}
+        .foldLeft(StatusCheck(status = false, None))((found, entry) => {
+        entry match {
+          case (_, kibanaCode) if kibanaCode.contains(s"Microservice(\"$serviceName\"") =>
+            StatusCheck(status = true, Some("https://github.com/hmrc/kibana-dashboards/tree/main/src/main/scala/uk/gov/hmrc/kibanadashboards/digitalservices"))
+          case _  =>
+            found
+        }
+      })
+    } yield statusCheck
+  }
+
+  private def checkGrafanaArchiveExists(serviceName: String)(implicit hc: HeaderCarrier): Future[StatusCheck] = {
+    for {
+      inputStream <- gitHubConnector.streamGithubCodeLoad("/hmrc/grafana-dashboards/zip/refs/heads/main")
+      zip = new ZipInputStream(inputStream.get)
+      statusCheck =  Iterator.continually(zip.getNextEntry)
```

Comment:
This is duplicated - think it' requires a helper util. It will be easier to read the logic if separated from the manual zip handling too.